File: javascript/challenge18.js
const countTickets = (tickets) => {
    let object = { red: 0, green: 0, blue: 0 };
    for (let i = 0; i < tickets.length; i++) {
        if (tickets[i] === 'red') {
            object['red']++;
        } else if (tickets[i] === 'green') {
            object['green']++;
        } else if (tickets[i] === 'blue') {
            object['blue']++;
        }
    }
    return object;
};

const bestOdds = (tickets, raffleEntries) => {
    let tally = countTickets(tickets)
    let rT = Object.keys(tally).map(i => tally[i])
    let rE = Object.keys(raffleEntries).map(i => raffleEntries[i])
    let result = rT.map(function (n, i) { return n / rE[i]; });
    console.log(result)
    if (result[0] >= result[1]) {
        if (result[2] > result[0]) {
            return "You have the best odds of winning the blue raffle."
        }
        return "You have the best odds of winning the red raffle."
    } else {
        if (result[2] >= result[1]) { 
            return "You have the best odds of winning the blue raffle."
        } else {
          return "You have the best odds of winning the green raffle."
        }
    }
};

File: python/challenge18.py
def countTickets(tickets):
    object = { 'red': 0, 'green': 0, 'blue': 0 }
    for i in range(0, len(tickets)):
        if (tickets[i] == 'red'):
            object['red'] = object['red'] + 1
        elif (tickets[i] == 'green'):
            object['green'] = object['green'] + 1
        elif (tickets[i] == 'blue'):
            object['blue'] = object['blue'] + 1
    return object

def bestOdds(tickets, raffleEntries): 
    tally = countTickets(tickets)
    red = 'Red'
    print(tally)
    return f'You have the best odds of winning the {red} raffle.'

tickets = [ 'red', 'red', 'green', 'blue', 'green' ]
raffleEntries = { 'red': 10, 'green': 30, 'blue': 15 }
bestOdds(tickets, raffleEntries)
